src/Extensions/Diagnostics.HealthChecks/Internal/HealthCheckPublisher.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.ObjectPool;
+using ServiceFabricHealth = System.Fabric.Health;
+
+namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
+{
+	internal abstract class HealthCheckPublisher : IHealthCheckPublisher
+	{
+		protected readonly ObjectPool<StringBuilder> StringBuilderPool;
+
+		protected string HealthReportSourceId { get { return HealthReportSourceIdImpl; } }
+
+		protected abstract string HealthReportSourceIdImpl { get; }
```

Comment:
Why do we need two separate properties? I think just making `HealthReportSourceId` abstract would give the same result?

Reply:
it was probably some workaround - I dropped this now.